flink-runtime/src/main/java/org/apache/flink/runtime/client/JobClient.java

```diff
@@ -275,9 +275,35 @@ public static ClassLoader retrieveClassLoader(
 	public static JobExecutionResult awaitJobResult(JobListeningContext listeningContext) throws JobExecutionException {
 
 		final JobID jobID = listeningContext.jobID;
+		final ActorRef jobClientActor = listeningContext.jobClientActor;
 		final Future<Object> jobSubmissionFuture = listeningContext.jobResultFuture;
 		final ClassLoader classLoader = listeningContext.classLoader;
 
+		while (!jobSubmissionFuture.isCompleted()) {
+			try {
+				Thread.sleep(250);
```

Comment:
Actually, it's not quite as easy. If we simply wait on the ask timeout via `Await.ready/result`, then we complete the `jobSubmissionFuture` with a timeout (if the job runs longer then the timeout interval). Subsequent checks will always return the same result.

Thus, we probably need something like a sleep here.